### ISSUE TITLE
Memory leak when listen fails

### ISSUE THREAD

**Issue by fatduckling** [uNetworking/uWebSockets#1085]
Hi
Thanks so much for the library.
I think I just found a memory leak when `token` in `listen(9001, [](auto *token)` is `NULL`. If it's null, then you can't close it with `us_listen_socket_close` to free memory. I intentionally set `key_file_name` to `/dev/null` so `listen` fails and I ran this against valgrind.

Test case:
```c++
#include <iostream>
#include "uWebSockets/App.h"

int main() {
  struct PerSocketData {};
  uWS::SSLApp(
      {
          .key_file_name = "/dev/null", // will fail
          .cert_file_name = "/dev/null",
          .passphrase = "1234"
      }
  ).ws<PerSocketData>("/*", {
      .open = [](auto *ws) {},
      .message = [](auto *ws, std::string_view message, uWS::OpCode opCode) {},
      .close = [](auto *ws, int code, std::string_view message) { }
  }).listen(9001, [](auto *token) {
    if (token) {
      std::cout << "Listening on port " << 9001 << std::endl;
    } else {
      std::cout << "Failed to listen on port. Token is null? " << std::boolalpha << (token == NULL) << std::endl;
    }
  }).run();
}

```

Valgrind errors:
```
> valgrind --leak-check=full ./ZMQ1 
==3563== Memcheck, a memory error detector
==3563== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==3563== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==3563== Command: ./ZMQ1
==3563== 
Failed to listen on port. Token is null? true
==3563== 
==3563== HEAP SUMMARY:
==3563==     in use at exit: 7,609 bytes in 27 blocks
==3563==   total heap usage: 4,845 allocs, 4,818 frees, 1,334,290 bytes allocated
==3563== 
==3563== 7,609 (3,808 direct, 3,801 indirect) bytes in 1 blocks are definitely lost in loss record 27 of 27
==3563==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==3563==    by 0x144111: us_create_socket_context (context.c:122)
==3563==    by 0x1451B1: us_internal_create_ssl_socket_context (openssl.c:403)
==3563==    by 0x1440A5: us_create_socket_context (context.c:111)
==3563==    by 0x11F5C2: uWS::HttpContext<true>::create(uWS::Loop*, us_socket_context_options_t) (HttpContext.h:340)
==3563==    by 0x11BFC2: uWS::TemplatedApp<true>::TemplatedApp(us_socket_context_options_t) (App.h:82)
==3563==    by 0x10E8AF: main (main.cpp:6)
==3563== 
==3563== LEAK SUMMARY:
==3563==    definitely lost: 3,808 bytes in 1 blocks
==3563==    indirectly lost: 3,801 bytes in 26 blocks
==3563==      possibly lost: 0 bytes in 0 blocks
==3563==    still reachable: 0 bytes in 0 blocks
==3563==         suppressed: 0 bytes in 0 blocks
==3563== 
==3563== For lists of detected and suppressed errors, rerun with: -s
==3563== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
```

I am using v0.5.0 uSockets, v18.10.0 uWebSockets and gcc version 9.3.0 (Ubuntu 9.3.0-10ubuntu2).

Thanks

**Comment by alexhultman** [comment]
Thanks for the excellent report. I'm working on some testing that will find these cases better.

**Comment by fatduckling** [comment]
Hi Alex,
Please see pull request above when you get the chance. Cheers

**Comment by alexhultman** [comment]
This is fixed in upcoming uSockets 0.6.0
